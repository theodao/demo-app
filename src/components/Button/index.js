import React from "react";
import styled from "styled-components";

const Button = styled.button`
  padding: 4px 12px;
  display: inline-block;
  cursor: pointer;
  border: 1px solid #bbb;
  overflow: visible;
  color: #fff;
  text-shadow: 0 1px 0
  white-space: nowrap;
  border-radius: 5px;
  background-color: #269CE9;
  border-color: #269CE9
 &:hover{
  background-color: #70B9E8;
  color: #555;
}

&:active{
  background: #269CE9;
  position: relative;
  top: 1px;
  text-shadow: none;
  box-shadow: 0 1px 1px rgba(0, 0, 0, .3) inset;
}
`;

export default ({ onClick = () => {}, children }) => {
  return <Button onClick={onClick}>{children}</Button>;
};
