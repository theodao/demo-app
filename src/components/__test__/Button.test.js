import React from "react";
import renderer from "react-test-renderer";
import Button from "../Button";

it("renders Button", () => {
  const tree = renderer.create(<Button />).toJSON();
  expect(tree).toMatchSnapshot();
});