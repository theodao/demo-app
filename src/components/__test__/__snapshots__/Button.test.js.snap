// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Button 1`] = `
<button
  className="sc-AxjAm gYogjw"
  onClick={[Function]}
/>
`;
